requirements.txt

```
@@ -17,6 +17,6 @@ openai>=1.13.3,<2.0.0
 sentencepiece>=0.2.0
 tabulate>=0.9.0
 tenacity>=8.3.0,!=8.4.0
-torch>=2.3.0,<2.5.0
+torch>=2.3.0,<2.6.0
```

Shouldn't this be
```suggestion
torch>=2.5.0,<2.6.0
```

...to enforce the use of 2.5? If other required packages in this file change (or their versions change), then pip may end up pulling 2.3.z or 2.4.z when trying to automatically resolve dependencies.

---

Having an older version is fine if that's a requirement for one component. That's unlikely to happen though, unless a library is downgrading its dependencies. And that's also what we've done for the instructlab/instructlab repo: https://github.com/instructlab/instructlab/pull/2865.

---

Gotcha. So we're allowing 2.5 if other packages support it, but not enforcing the use of 2.5.

I was confused by the PR title because it suggested we were actually updating the version.

---

Thanks for hightling this @courtneypacheco ! Looking in the CI logs, even though this raises the cap it looks like CI is actually installing torch 2.4.x and running with that. So, we may have more work to do here to figure out what dependency is forcing our torch to 2.4.x instead of 2.5.x.

---

Ahh, I see in the logs:
```
2025-01-08T12:55:07.4571834Z Collecting torch<2.5.0,>=2.3.0 (from instructlab==0.21.0a2.dev258)
2025-01-08T12:55:07.4574409Z   Obtaining dependency information for torch<2.5.0,>=2.3.0 from 
```

Our CI tests install instructlab from latest main, so we must have picked up instructlab before the torch bump was merged there. Kicking this off again to ensure it picks up torch 2.5.x now.

---

I suspect that it's because we're still on vLLM 0.6.2, which depends on PyTorch 2.4.

---

Ahh, yes. So in other words, neither instructlab/instructlab nor instructlab/sdg can actually bump our torch version in use until we move to a newer vllm than 0.6.2 that depends on torch 2.5.x. Regardless of this version range, our vllm dependency specifies an exact dependency on torch 2.4.0.

---

Yes, that seems to be the case. I have opened a new PR in instructlab to update vLLM to 0.6.6.post1: https://github.com/instructlab/instructlab/pull/2892.